presto-verifier/src/main/java/com/facebook/presto/verifier/Validator.java

```diff
@@ -341,29 +365,44 @@ private QueryResult executeQueries(
             }
         }
         finally {
-            int retry = 0;
-            QueryResult tearDownResult;
-            do {
-                tearDownResult = tearDown(query, postQueryResults, postQuery -> executeQueryInternal(gateway, username, password, query, postQuery, timeout, sessionProperties));
-                if (tearDownResult.getState() == State.SUCCESS) {
-                    break;
-                }
-                try {
-                    TimeUnit.MINUTES.sleep(1);
-                }
-                catch (InterruptedException e) {
-                    Thread.currentThread().interrupt();
-                    break;
-                }
-                retry++;
+            if (dropTemporaryTables) {
```

Comment:
The config option is for dropping on a checksum failure, but we still want to drop tables if the query fails.

Reply:
This is a good point. Now the validator will not drop tables when executing, and will only drop them when `validate()` returns if necessary (the only case that these tables will be kept is where checksum fails and the config is set to false).